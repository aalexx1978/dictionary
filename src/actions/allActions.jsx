export const FETCH_DATA = "FETCH_DATA";
export const RECEIVE_PRODUCTS = "RECEIVE_PRODUCTS";
export const RECEIVE_DICTIONARIES = "RECEIVE_DICCIONARIES";
export const ERROR = "ERROR";
export const SET_DICTIONARY = "SET_DICTIONARY";
