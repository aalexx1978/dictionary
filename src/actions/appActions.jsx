import axios from "axios";
import {
    RECEIVE_PRODUCTS,
    RECEIVE_DICTIONARIES,
    ERROR,
    SET_DICTIONARY
} from "./allActions";

const headers = { "Content-Type": "application/json" };
export function receiveProducts(data) {
    return { type: RECEIVE_PRODUCTS, products: data };
}
export function receiveDictionaries(data) {
    return { type: RECEIVE_DICTIONARIES, dictionaries: data };
}
export function isError(errorValue) {
    return { type: ERROR, error: errorValue };
}

export function setDictionary(value) {
    return { type: SET_DICTIONARY, dictionary: value };
}

export function fetchProducts() {
    return dispatch => {
        axios
            .get("/products")
            .then(function(response) {
                dispatch(receiveProducts(response.data));
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}
export function addProduct(data) {
    let content = {
        product: "Sony X",
        color: "Mystic Silver",
        price: 772
    };
    return dispatch => {
        axios
            .post("/products", content, headers)
            .then(function(response) {
                dispatch(fetchProducts());
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}
export function addDictionary(data) {
    let content = {
        name: "New Dictionary",
        Stonegrey: "Dark",
        "Midnight Black": "Black Dark",
        "Mystic Silver": "Silver Special"
    };
    return dispatch => {
        axios
            .post("/dictionaries", content, headers)
            .then(function(response) {
                dispatch(fetchDictionaries());
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}
export function removeDictionary(id) {
    return dispatch => {
        axios
            .delete("/dictionaries/" + id)
            .then(function(response) {
                dispatch(fetchDictionaries());
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}
export function removeProduct(id) {
    return dispatch => {
        axios
            .delete("/products/" + id)
            .then(function(response) {
                dispatch(fetchProducts());
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}

export function updateDictionary(dic) {
    return dispatch => {
        axios
            .put("/dictionaries/" + dic.id, dic, headers)
            .then(function(response) {
                dispatch(fetchDictionaries());
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}
export function fetchDictionaries() {
    return dispatch => {
        axios
            .get("/dictionaries")
            .then(function(response) {
                dispatch(receiveDictionaries(response.data));
            })
            .catch(function(error) {
                dispatch(isError(error.message));
                setTimeout(() => {
                    dispatch(isError(null));
                }, 4500);
            });
    };
}
