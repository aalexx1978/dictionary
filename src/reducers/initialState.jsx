export default {
    products: [],
    dictionaries: [],
    error: null,
    dictionary: 0
};
