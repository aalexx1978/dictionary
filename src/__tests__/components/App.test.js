import React from "react";
import Enzyme from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import initialState from "../../reducers/initialState";

import * as actions from "../../actions/appActions";
import ConnectedApp, { App } from "../../components/App";

Enzyme.configure({ adapter: new Adapter() });
describe(">>>APP --- Snapshot", () => {
    it("+++capturing Snapshot of App", () => {
        const renderedValue = renderer
            .create(<App {...initialState} appActions={actions} />)
            .toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
