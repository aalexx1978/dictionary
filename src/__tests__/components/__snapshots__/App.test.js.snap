// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>APP --- Snapshot +++capturing Snapshot of App 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <nav
          className="navbar navbar-default"
        >
          <div
            className="container"
          >
            <div
              className="navbar-header"
            >
              <span
                className="navbar-brand"
              >
                <img
                  alt="logo"
                  className="App-logo"
                  src="logo.png"
                />
                Dictionary
              </span>
              <button
                className="navbar-toggle collapsed"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="sr-only"
                >
                  Toggle navigation
                </span>
                <span
                  className="icon-bar"
                />
                <span
                  className="icon-bar"
                />
                <span
                  className="icon-bar"
                />
              </button>
            </div>
          </div>
        </nav>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
            htmlFor="formControlsSelect"
          >
            Select Dictionary to apply
          </label>
          <select
            className="form-control"
            defaultValue={0}
            id="formControlsSelect"
            onChange={[Function]}
            placeholder="select"
          >
            <option
              value={0}
            >
              Original
            </option>
          </select>
        </div>
      </div>
      <div
        className="col-md-2"
      >
        <label
          className="control-label"
        >
          Add new product
        </label>
        <button
          className="btn btn-info"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add Product
        </button>
      </div>
    </div>
    <div
      className="row"
    />
    <div
      className="row"
    />
    <div
      className="row"
    >
      <div
        className="col-md-3"
      >
        <button
          className="btn btn-warning"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add Dictionary
        </button>
      </div>
    </div>
    <div
      className="btm row"
    >
      <div
        className="col-xs-7"
      >
        <div
          className="error1"
        >
           
        </div>
         Warning, domain or range repeated
      </div>
      <div
        className="col-xs-5"
      >
        <div
          className="error2 danger"
        >
           
        </div>
         Error, possible bucle
      </div>
    </div>
  </div>
</div>
`;
